src/audit/unsecure_commands_allowed.rs

```diff
@@ -0,0 +1,117 @@
+
+pub(crate) struct UnsecureCommandsAllowed;
+
+impl UnsecureCommandsAllowed {
+    fn insecure_commands_allowed<'w>(
+        &self,
+        workflow: &'w Workflow,
+        location: SymbolicLocation<'w>,
+    ) -> Finding<'w> {
+        FindingBuilder::new(ID, DESCRIPTION)
+            .confidence(Confidence::High)
+            .severity(Severity::High)
+            .add_location(
+                location
+                    .with_keys(&["env".into()])
+                    .annotated("insecure GHA commands allowed here"),
+            )
+            .build(workflow)
+            .expect("Cannot build a Finding instance")
+    }
+
+    fn has_unsecure_commands_enabled(&self, env: &Env) -> bool {
+        if let Some(value) = env.get("ACTIONS_ALLOW_UNSECURE_COMMANDS") {
+            // Simple emulation for type coercion from YAML declared values at GHA runtime
+            match value {
+                EnvValue::String(string) => !string.is_empty(),
+                EnvValue::Number(number) => *number > 0.0,
+                EnvValue::Boolean(it) => *it,
+            }
+        } else {
+            false
+        }
+    }
+
+    fn audit_steps<'w>(&self, workflow: &'w Workflow, steps: Steps<'w>) -> Vec<Finding<'w>> {
+        steps
+            .into_iter()
+            .filter(|step| {
+                let StepBody::Run {
+                    run: _,
+                    working_directory: _,
+                    shell: _,
+                    ref env,
+                } = &step.deref().body
+                else {
+                    return false;
+                };
+
+                self.has_unsecure_commands_enabled(env)
+            })
+            .map(|step| self.insecure_commands_allowed(workflow, step.location()))
+            .collect()
+    }
+}
+
+impl WorkflowAudit for UnsecureCommandsAllowed {
+    fn ident() -> &'static str
+    where
+        Self: Sized,
+    {
+        ID
+    }
+
+    fn desc() -> &'static str
+    where
+        Self: Sized,
+    {
+        DESCRIPTION
+    }
+
+    fn new(_: AuditState) -> anyhow::Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(Self {})
+    }
+
+    fn audit<'w>(&self, workflow: &'w Workflow) -> anyhow::Result<Vec<Finding<'w>>> {
+        let mut results = vec![];
+
+        if self.has_unsecure_commands_enabled(&workflow.env) {
+            results.push(self.insecure_commands_allowed(workflow, workflow.location()))
+        }
+
+        for job in workflow.jobs() {
+            match *job {
+                Job::NormalJob(normal) => {
+                    if self.has_unsecure_commands_enabled(&normal.env) {
+                        results.push(self.insecure_commands_allowed(workflow, job.location()))
+                    }
+
+                    results.extend(self.audit_steps(workflow, job.steps()))
+                }
+                Job::ReusableWorkflowCallJob(reusable) => {
+                    if self.has_unsecure_commands_enabled(&reusable.with) {
```

Comment:
I *think* this might be wrong: a reusable workflow's `with:` clause can only contain inputs, not environment variables. So I think we can skip these entirely, and handle only normal jobs :slightly_smiling_face:

Reply:
I missed that! Fixed [here](https://github.com/woodruffw/zizmor/pull/176/commits/f35682c69214f1f8e067af61d80acd434308f288)